gcpdiag/queries/cloudsql.py

```diff
@@ -67,6 +67,14 @@ def has_maint_window(self) -> int:
     except KeyError:
       return 0
 
+  @property
+  def has_del_protection(self) -> bool:
+    try:
+      return get_path(self._resource_data,
+                      ('settings', 'deletionProtectionEnabled'))
+    except KeyError:
```

Comment:
get_path doesn't throw an exception I think. You can give it a default value if the key is not found:

    return get_path(self._resource_data,
                      ('settings', 'deletionProtectionEnabled')), False)

Reply:
Done